### PR TITLE
Add GrantPeriodicCondition from AS.

### PR DESCRIPTION
**Pull request by GraionDilach**

### CODE REVIEW

File: OpenRA.Mods.HV/Traits/Conditions/GrantPeriodicCondition.cs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+		}
+
+		float ISelectionBar.GetValue()
+		{
+			if (!info.ShowSelectionBar)
+				return 0f;
+
+			return IsEnabled
+				? (float)(active - ticks) / active
+					: (float)ticks / cooldown;
```
Comment (GraionDilach) [inline OpenRA.Mods.HV/Traits/Conditions/GrantPeriodicCondition.cs:165]: Derp.

File: OpenRA.Mods.HV/Traits/Conditions/GrantPeriodicCondition.cs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+			this.info = info;
+		}
+
+		void SetDefaultState()
+		{
+			if (info.StartsGranted)
+			{
+				ticks = info.ActiveDuration.Length == 2
+					? self.World.SharedRandom.Next(info.ActiveDuration[0], info.ActiveDuration[1])
+					: info.ActiveDuration[0];
```
Comment (Mailaender) [inline OpenRA.Mods.HV/Traits/Conditions/GrantPeriodicCondition.cs:68]: ticks = Common.Util.RandomInRange(self.World.SharedRandom, info.ActiveDuration);

Comment (GraionDilach) [reply OpenRA.Mods.HV/Traits/Conditions/GrantPeriodicCondition.cs:68]: Oh, this looks new to me. Applied to both Active and Cooldown.

Comment (Mailaender) [reply OpenRA.Mods.HV/Traits/Conditions/GrantPeriodicCondition.cs:68]: Still missing here.

Comment (GraionDilach) [reply OpenRA.Mods.HV/Traits/Conditions/GrantPeriodicCondition.cs]: Right, this was in `SetDefaultState`... nvm.

**Pull request merged**